--- src/hdmi.h.orig	2014-03-19 23:46:15.121494476 +0100
+++ src/hdmi.h	2014-03-19 23:58:32.232227739 +0100
@@ -35,10 +35,11 @@
         CEC::cec_logical_address logical;
         uint8_t port;
     };
-};
+    
+    std::ostream& operator<<(std::ostream &out, const HDMI::physical_address & address);
+    std::istream& operator>>(std::istream &in, HDMI::physical_address & address);
 
-std::ostream& operator<<(std::ostream &out, const HDMI::physical_address & address);
-std::istream& operator>>(std::istream &in, HDMI::physical_address & address);
+    std::istream& operator>>(std::istream &in, HDMI::address & address);
+    std::ostream& operator<<(std::ostream &out, const HDMI::address & address);
+};
 
-std::ostream& operator<<(std::ostream &out, const HDMI::address & address);
-std::istream& operator>>(std::istream &in, HDMI::address & address);
--- src/hdmi.cpp.orig	2014-03-20 00:03:00.318857434 +0100
+++ src/hdmi.cpp	2014-03-20 00:03:08.428755478 +0100
@@ -6,13 +6,13 @@
 
 #include <cstdio>
 
-std::ostream& operator<<(std::ostream &out, const HDMI::physical_address & address)
+std::ostream& HDMI::operator<<(std::ostream &out, const HDMI::physical_address & address)
 {
     return out << address[0] << '.' << address[1] << '.'
                << address[2] << '.' << address[3];
 }
 
-std::istream& operator>>(std::istream &in, HDMI::physical_address & address)
+std::istream& HDMI::operator>>(std::istream &in, HDMI::physical_address & address)
 {
     std::string s;
 
@@ -56,7 +56,7 @@
     return in;
 }
 
-std::istream& operator>>(std::istream &in, HDMI::address & address)
+std::istream& HDMI::operator>>(std::istream &in, HDMI::address & address)
 {
     char c = in.peek();
     if( c >= '0' && c <= '9' )
@@ -116,7 +116,7 @@
     return in; 
 }
 
-std::ostream& operator<<(std::ostream &out, const HDMI::address & address)
+std::ostream& HDMI::operator<<(std::ostream &out, const HDMI::address & address)
 {
     switch( address.logical )
     {
